doc/wallet-storage.md

```diff
@@ -0,0 +1,255 @@
+# Wallet Storage Design
+
+In current state libindy allows to plug different wallet implementations. Plugged wallet now handles both security
+and storage layers. This design proposes to restrict our plugged interface by handling only storage layer.
+All encryption will be performed in libindy. It will simplify plugged wallets and warranty good security level
+for 3d party wallets implementations.
+
+Also proposals enhances our API for efficient and flexible search with paging support.
+
+## Goals and ideas
+
+* Simplify plugged wallets and warranty security level for 3d party wallets implementation
+* Perform all encryption on libindy level:
```

Comment:
What keys will be used for encryption? Where are they persisted? How libindy has access to them? Are there different keys, or one fey for the whole wallet?